ec2/sc-ec2-windows-jumpcloud.yaml

```diff
@@ -0,0 +1,323 @@
+                  - ' -JcServiceApiKey '
+                  - Fn::Transform: {"Name": "SsmParam", "Parameters": {"Type": "SecureString", "Name": "/infra/JcServiceApiKey"}}
+                  - ' -JcSystemsGroupId '
+                  - Fn::Transform: {"Name": "SsmParam", "Parameters": {"Type": "SecureString", "Name": "/infra/JcSystemsGroupId"}}
+                  - ' -OwnerEmail '
+                  - !Ref OwnerEmail
+                  - ' > C:\scripts\associate-jc-system.log'
+    Properties:
+      ImageId: !Ref 'LatestAmiId'
+      InstanceType: !Ref 'WindowsInstanceType'
+      SubnetId: !ImportValue
+        'Fn::Sub': '${AWS::Region}-internalpoolvpc-PrivateSubnet1'
+      SecurityGroupIds:
+        - !Ref 'WindowsSecurityGroup'
+      KeyName: !Ref 'KeyPair'
+      IamInstanceProfile: !Ref 'PatchingInstanceProfile'
+      UserData:
+        Fn::Base64: !Sub |
+          <script>
+          cfn-init.exe -v --stack ${AWS::StackId} --resource WindowsInstance --region ${AWS::Region} --configsets SetupCfn,SetupApps,SetupJumpcloud
+          cfn-signal.exe -e %errorlevel% --stack ${AWS::StackId} --resource WindowsInstance --region ${AWS::Region}
+          </script>
+      Tags:
+        - Key: Name
+          Value: SC-EC2-RA-Windows-Instance
+        - Key: Description
+          Value: Service-Catalog-EC2-Reference-Architecture
+  WindowsPatchBaseline:
+    Type: AWS::SSM::PatchBaseline
+    Properties:
+      OperatingSystem: WINDOWS
+      ApprovalRules:
+        PatchRules:
+          - ApproveAfterDays: 0
+            ComplianceLevel: CRITICAL
+            PatchFilterGroup:
+              PatchFilters:
+                - Key: PRODUCT
+                  Values:
+                    - WindowsServer2016
+                - Key: CLASSIFICATION
+                  Values:
+                    - SecurityUpdates
+                - Key: MSRC_SEVERITY
+                  Values:
+                    - Critical
+                    - Important
+                    - Moderate
+                    - Low
+                    - Unspecified
+      Description: Service Catalog EC2 Reference Architecture Patch Baseline for Microsoft
+        Windows instace
+      Name: sc-ec2-ra-windows-patch-baseline
+  MaintenanceWindowRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AmazonSSMMaintenanceWindowRole
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+                - ssm.amazonaws.com
+            Action:
+              - sts:AssumeRole
+  MaintenanceWindow:
+    Type: AWS::SSM::MaintenanceWindow
+    Properties:
+      Description: Maintenance window to allow for patching windows instances
+      AllowUnassociatedTargets: false
+      Cutoff: 2
+      Schedule: cron(* 17 * * ? *)
```

Comment:
I think this schedule is a good compromise between updates and availability... We need the machines to be up, but we also don't want to force reboots on people. An alternative would be to define `ScheduleTimezone` explicitly and set `Schedule` to the middle of the night, but its not clear to that users will park or not. Anyway, I defer to you if you think this gets the best coverage for the user base.